### ISSUE TITLE
Package not compatible with newer versions of React Native: "bundleReleaseAar" fails

### ISSUE THREAD

**Issue by andersonaddo** [cjam/react-native-spotify-remote#194]
`./gradlew :app:assembleRelease`(suggested [here](https://github.com/cjam/react-native-spotify-remote/issues/90#issuecomment-819091409)) doesn't work for me.

@cjam you mentioned in #57 that you were able to build and release; was that from the `/example` app? You mentioned you did it in Android Studio, but I wonder if you have tried from the cli using `app:assembleRelease`. I ask because I wonder if this is a gradle version issue.

The example [uses a pretty old version of gradle](https://github.com/cjam/react-native-spotify-remote/blob/5a4b4aee407b6239638775e2155842f5be48f2af/example/android/build.gradle#L15), newer RN apps that are created now use 7+

May investigate myself if time permits.

Edit: I'm [pretty sure](https://stackoverflow.com/questions/71268813/direct-local-aar-file-dependencies-are-not-supported-when-building-an-aar-reac) it is a gradle version issue. Would love to see this fixed, because it means the package isn't gonna work for newest versions of RN.

Edit 2: You can see the bump in gradle between the `/example` and new RN projects [here](https://react-native-community.github.io/upgrade-helper/?from=0.63.4&to=0.68.2) (Ctrl F for android/build.gradle).

_Originally posted by @andersonaddo in https://github.com/cjam/react-native-spotify-remote/issues/90#issuecomment-1135352375_

**Comment by cjam** [comment]
@andersonaddo thanks for the investigation.

Yea, I had started with the iOS side of this library as I had an iOS device, then someone kindly contributed an Android implementation back which forced me to learn about android.. but clearly I've got some gaps.  

Are you able to update the gradle version and issue a PR?  I can also do it, but it will just take longer for me to find time as work is busy these days.

**Comment by andersonaddo** [comment]
The issue has to do with how the Android native side deals with .aar dependencies, which is something I am completely unfamiliar with 😂. 

Though I suppose there are plenty of guides online, but to be frank I doubt lll get to it (I decided to downgrade my projects RN version instead)

**Comment by cjam** [comment]
Ah gotcha, yea it seems like it should be straightforward.  I just don't have the capacity to dig into it at the moment.

**Comment by gustavoggs** [comment]
@cjam @andersonaddo I sent a PR to solve that issue. I it will work with all versions and we can think in a more generic solution if needed.

**Comment by theshadowagent** [comment]
This is still an issue for me on the latest `master` commit (7c13126ec6ce5e990a83aa730b65382ce709e37a), and can confirm I'm using `./gradlew :app:assembleRelease`, and that debug build is working.

Trying the versions from PR #199 & #205 didn't help.
The latest merged PR #205 produced the same issue as described here, and PR #199 by @gustavoggs produces a new issue

```
[RUN_GRADLEW] WARNING:Using flatDir should be avoided because it doesn't support any meta-data formats.
[RUN_GRADLEW] FAILURE: Build failed with an exception.
[RUN_GRADLEW] * What went wrong:
[RUN_GRADLEW] Could not determine the dependencies of task ':app:lintVitalReportRelease'.
[RUN_GRADLEW] > Could not resolve all task dependencies for configuration ':app:releaseRuntimeClasspath'.
[RUN_GRADLEW]    > Could not find :spotify-auth-release-1.2.3:.
[RUN_GRADLEW]      Required by:
[RUN_GRADLEW]          project :app > project :react-native-spotify-remote
[RUN_GRADLEW]    > Could not find :spotify-app-remote-release-0.7.2:.
[RUN_GRADLEW]      Required by:
[RUN_GRADLEW]          project :app > project :react-native-spotify-remote
```

**Comment by gijsfonk** [comment]
Hi there,

I too struggled with this issue after updating my project to te latests React Native version (0.71.0). I had hope that the latests PRs would solve the issue, just like @theshadowagent, but alas.

However, upon researching the issue I tried numerous "solutions" and I think I have found a working one (at least for me it does). 

[This Stackoverflow answer](https://stackoverflow.com/a/70074787/17387374) seems to have solved the issue. I'm not sure if it's the "best way" of solving this .aar issue, but it works and I haven't found an other solution yet, so...

In short, it requires you to add the .aar files as a module to your main Android project folder (instead of the one in this plugin).

Hope this is helpful.

For reference, I'm using the following versions:
```
RN: 0.71.0
Spotify remote (this lib): 1.0.0-1
Android Gradle Plugin: 7.2.2
Gradle version: 7.5.1
Gradle JDK: (Android Studio) Embedded JDK version 11.0.15
Android Studio: 2022.1.1
```

**Comment by remiilekun** [comment]
Hello @gijsfonk, please do you have an example repo or gists of the stack overflow solution implementation? I followed the steps but my build is still failing.

**Comment by CaptainHaider** [comment]
Hello @gijsfonk, I followed but my build is also failing, can you please provide some example how did you solve the issue.

**Comment by theshadowagent** [comment]
Here is my fork I ended up using, might open a PR soon with some explanations
https://github.com/theshadowagent/react-native-spotify-remote/commits/fork
